docs/release-process-deep-dive.md

```diff
@@ -0,0 +1,596 @@
+```
+</details>
+
+The following sections are not strictly in the same order as the workflow but intended to bucket the different responsibilities.
+
+### Site Manual
+
+A git commit is created in the `cli.github.com` site repository containing the contents of the CLI Manual uploaded by the [`linux`](#linux) job. This is not pushed until the package repository artifacts are set up later.
+
+### Site Package Repositories
+
+The `cli.github.com` website hosts RPM and Debian package repositories to support the [official sources installation instructions](https://github.com/cli/cli/blob/trunk/docs/install_linux.md#official-sources). In order to provide a secure installation method, artifacts in these repositories are signed by a GPG key, which must be loaded into `gpg` for use in later steps.
+
+```sh
+base64 -d <<<"$GPG_PUBKEY" | gpg --import --no-tty --batch --yes
+base64 -d <<<"$GPG_KEY" | gpg --import --no-tty --batch --yes
+echo "allow-preset-passphrase" > ~/.gnupg/gpg-agent.conf
+gpg-connect-agent RELOADAGENT /bye
+/usr/lib/gnupg2/gpg-preset-passphrase --preset "$GPG_KEYGRIP" <<<"$GPG_PASSPHRASE"
+```
+
+> [!NOTE]
+> TODO: Further Documentation on these lines.
+
+#### RPM
+
+The `.rpm` files uploaded by the [`linux`](#linux) job are signed using [`rpmsign`](https://man7.org/linux/man-pages/man8/rpmsign.8.html). The [`createrepo`](https://linux.die.net/man/8/createrepo) tool is used to generate a `repomd.xml` metadata file which describes the contents of a Red Hat repository. The artfiacts and `repomd.xml` file are then copied into the site repository, and the `repomd.xml` is signed using `gpg --yes --detach-sign --armor repodata/repomd.xml`, producing a signature file.
+
+> [!WARNING]
+> The `createrepo` tool is executed inside a [Docker container](https://github.com/cli/cli/blob/756f4ec04abdc9fdbab3fef35b182c546ef1dd17/script/createrepo.sh) for [package management reasons](https://github.com/cli/cli/pull/2856) that may no longer be true.
+
+#### Debian
+
+The `.deb` files uplaoded by the [`linux`](#linux) job are iterated per Debian release (see warning below), using [`reprepro`](https://manpages.debian.org/bookworm/reprepro/reprepro.1.en.html) which produces a directory and file structure for a Debian package repository. The [`./script/distributions`](https://github.com/cli/cli/blob/756f4ec04abdc9fdbab3fef35b182c546ef1dd17/script/distributions) `SignWith` lines indicate the GPG Key ID that `reprepro` should use to sign packages in the created repository. The generated directories are then copied to the site repository.
+
+> [!WARNING]
+> There is a note that we should remove [legacy distributions from our list](https://github.com/cli/cli/blob/756f4ec04abdc9fdbab3fef35b182c546ef1dd17/.github/workflows/deployment.yml#L329-L332) but no indication of when that would happen.
+
+### Attest Artifacts
+
+Attestations are created for each of the release artifacts. For an example see: https://github.com/cli/cli/attestations/4920729
```

Comment:
Is there value pointing users towards https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds for more information?

Reply:
Sure.